src/index.ts

```diff
@@ -160,76 +238,96 @@ export const getNPMMetadataForDep = async (dep, npmAuthToken?: string) => {
-      warn(`${message}<br/><i>${idea}</i>`)
+      warn(renderMessage(msg, idea))
     }
   }
 }
 
 // Don't ship @types dependencies to consumers of Danger
-export const checkForTypesInDeps = packageDiff => {
+export const checkForTypesInDeps = (
+  packageDiff: JSONDiff,
+  { renderMessage = defaultRenderMessage }: Options = {},
+): void => {
   const sentence = danger.utils.sentence
 
   if (packageDiff.dependencies && packageDiff.dependencies.added) {
-    const typesDeps = packageDiff.dependencies.added.filter(d => d.startsWith("@types")).map(printDep)
+    const typesDeps = packageDiff.dependencies.added.filter(d => d.startsWith("@types")).map(escapePackage)
     if (typesDeps.length) {
-      const message = `@types dependencies were added to package.json, as a dependency for others.`
+      const msg = `@types dependencies were added to package.json, as a dependency for others.`
       const idea = `You need to move ${sentence(typesDeps)} into "devDependencies"?`
-      fail(`${message}<br/><i>${idea}</i>`)
+      fail(renderMessage(msg, idea))
     }
   }
 }
 
 export interface Options {
-  pathToPackageJSON?: string
+  pathToRootPackageJSON?: string
+  pathToPackageJSON?: string | ReadonlyArray<string>
   npmAuthToken?: string
+  renderPackageDetails?: typeof defaultRenderPackageDetails
+  renderReadme?: typeof defaultRenderReadme
+  renderYarnWhy?: typeof defaultRenderYarnWhy
+  renderMessage?: typeof defaultRenderMessage
+  renderPackageLink?: typeof defaultRenderPackageLink
 }
 
 /**
  * Provides dependency information on dependency changes in a PR
  */
-export default async function yarn(options: Options = {}) {
-  const path = options.pathToPackageJSON ? options.pathToPackageJSON : "package.json"
-  const packageDiff = await danger.git.JSONDiffForFile(path)
+export default async function yarn(options: Options = {}): Promise<void> {
+  const pathToRootPackageJSON = options.pathToRootPackageJSON ? options.pathToRootPackageJSON : "package.json"
+
+  let pathToPackageJSONs: ReadonlyArray<string>
+  if (options.pathToPackageJSON) {
+    pathToPackageJSONs = Array.isArray(options.pathToPackageJSON)
+      ? options.pathToPackageJSON
+      : [options.pathToPackageJSON]
+  } else {
+    const isNonRootPackageJson = (path: string) => path.endsWith("package.json") && path !== pathToRootPackageJSON
+
+    pathToPackageJSONs = Array.from(
+      new Set([...danger.git.created_files, ...danger.git.modified_files].filter(isNonRootPackageJson)),
+    )
+  }
+
+  const rootPackageDiff = await danger.git.JSONDiffForFile(pathToRootPackageJSON)
+  const packageDiffs: ReadonlyArray<JSONDiff> = await Promise.all(
+    pathToPackageJSONs.map(
+      (pathToPackageJSON: string): Promise<JSONDiff> => danger.git.JSONDiffForFile(pathToPackageJSON),
+    ),
+  )
 
-  checkForRelease(packageDiff)
-  checkForLockfileDiff(packageDiff)
-  checkForTypesInDeps(packageDiff)
+  checkForLockfileDiff(rootPackageDiff)
```

Comment:
I've just realised that I've made a mistake here. It's not only the root `package.json` which will result in `yarn.lock` changes.

I will need to rethink this and try to achieve this working within a workspace without any breaking changes to `yarn(options: Options)`.

- [ ] Make sure that all changes to `package.json` files result in a check of the `yarn.lock` file.
- [ ] Try to avoid a breaking change to `Options#pathToPackageJson`.